mllib/src/main/scala/org/apache/spark/mllib/clustering/PowerIterationClustering.scala

```diff
@@ -322,7 +324,11 @@ object PowerIterationClustering extends Logging {
       }, preservesPartitioning = true).cache()
     val sum = r.values.map(math.abs).sum()
     val v0 = r.mapValues(x => x / sum)
-    Graph(VertexRDD(v0), g.edges)
+    val graph = Graph(VertexRDD(v0), g.edges)
+    materialize(graph)
+    g.unpersist()
```

Comment:
`g` is a given `Graph` parameter. From the view of `randomInit`, it looks pretty weird to unpersist a given graph, before knowing it's persisted or not...

Reply:
Internally it persists its vertices and edges.

Reply:
oh, I see. Graph construction uses MEMORY_ONLY as default StorageLevel...